model_compression_toolkit/qunatizers_infrastructure/common/base_inferable_quantizer.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from enum import Enum
+
+
+class QuantizationTarget(Enum):
+    Activation = 0
+    Weights = 1
+
+
+class BaseInferableQuantizer:
```

Comment:
I think this name should be BaseQuantizer without the infereable, and BaseTrainableQuantizer and BaseInferableQuantizer should inherit from it.